services/src/test/java/edu/unc/lib/dl/cdr/services/processing/AddContainerIT.java

```diff
@@ -0,0 +1,187 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import edu.unc.lib.dl.acl.exception.AccessRestrictionException;
+import edu.unc.lib.dl.acl.service.AccessControlService;
+import edu.unc.lib.dl.acl.util.AccessGroupSet;
+import edu.unc.lib.dl.acl.util.GroupsThreadStore;
+import edu.unc.lib.dl.fcrepo4.AdminUnit;
+import edu.unc.lib.dl.fcrepo4.CollectionObject;
+import edu.unc.lib.dl.fcrepo4.ContentContainerObject;
+import edu.unc.lib.dl.fcrepo4.FolderObject;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectFactory;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.test.TestHelper;
+
+/**
+ *
+ * @author harring
+ *
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration({"/spring-test/test-fedora-container.xml", "/spring-test/cdr-client-container.xml", "/add-container-it-servlet.xml"})
+@WebAppConfiguration
+public class AddContainerIT {
+
+    @Autowired
+    private WebApplicationContext context;
+    @Autowired
+    private RepositoryObjectFactory repositoryObjectFactory;
+    @Autowired
+    private RepositoryObjectLoader repositoryObjectLoader;
+    @Autowired
+    private AccessControlService aclService;
+
+    private MockMvc mvc;
+
+    @Before
+    public void init() {
+        mvc = MockMvcBuilders
+                .webAppContextSetup(context)
+                .build();
+
+        TestHelper.setContentBase("http://localhost:48085/rest");
+
+        GroupsThreadStore.storeUsername("user");
+        GroupsThreadStore.storeGroups(new AccessGroupSet("adminGroup"));
+    }
+
+    @After
+    public void tearDown() {
+        GroupsThreadStore.clearStore();
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testAddCollectionToAdminUnit() throws UnsupportedOperationException, Exception {
+        PID parentPid = makePid();
+
+        AdminUnit parent = repositoryObjectFactory.createAdminUnit(parentPid, null);
+
+        assertChildContainerNotAdded(parent);
+
+        MvcResult result = mvc.perform(post("/edit/create/collection/" + parentPid.getUUID()))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn();
+
+        assertChildContainerAdded(parent);
+
+        // Verify response from api
+        Map<String, Object> respMap = getMapFromResponse(result);
+        assertEquals(parentPid.getUUID(), respMap.get("pid"));
+        assertEquals("create", respMap.get("action"));
+    }
+
+    @SuppressWarnings("unchecked")
```

Comment:
According to eclipse these suppressions aren't needed on any of the test methods in this class

Reply:
removed. funny, since i thought eclipse suggested these at one point. fickle IDE!